File: frame/src/main/java/com/dreampany/frame/data/source/repository/StoreRepository.java
package com.dreampany.frame.data.source.repository;

import com.dreampany.frame.data.model.Store;
import com.dreampany.frame.data.source.StoreDataSource;
import com.dreampany.frame.misc.ResponseMapper;
import com.dreampany.frame.misc.Room;
import com.dreampany.frame.misc.RxMapper;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

import hugo.weaving.DebugLog;
import io.reactivex.Maybe;

/**
 * Created by Hawladar Roman on 7/18/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */
@Singleton
public class StoreRepository extends Repository<Long, Store> implements StoreDataSource {

    private final StoreDataSource room;

    @DebugLog
    @Inject
    StoreRepository(RxMapper rx,
                    ResponseMapper rm,
                    @Room StoreDataSource room) {
        super(rx, rm);
        this.room = room;
    }

    @Override
    public List<String> getItemsOf(String type, String subtype, String state) {
        return room.getItemsOf(type, subtype, state);
    }

    @Override
    public Maybe<List<String>> getItemsOfRx(String type, String subtype, String state) {
        return room.getItemsOfRx(type, subtype, state);
    }

    @Override
    public Maybe<List<String>> getItemsOfRx(String type, String subtype, String state, int limit) {
        return room.getItemsOfRx(type, subtype, state, limit);
    }

    @Override
    public List<Store> getItems(String type, String subtype, String state) {
        return null;
    }

    @Override
    public Maybe<List<Store>> getItemsRx(String type, String subtype, String state) {
        return null;
    }

    @Override
    public boolean isEmpty() {
        return room.isEmpty();
    }

    @Override
    public Maybe<Boolean> isEmptyRx() {
        return Maybe.fromCallable(this::isEmpty);
    }

    @Override
    public int getCount() {
        return room.getCount();
    }

    @Override
    public Maybe<Integer> getCountRx() {
        return room.getCountRx();
    }

    @Override
    public boolean isExists(Store store) {
        return false;
    }

    @Override
    public Maybe<Boolean> isExistsRx(Store store) {
        return null;
    }

    @Override
    public long putItem(Store store) {
        return room.putItem(store);
    }

    @Override
    public Maybe<Long> putItemRx(Store store) {
        return null;
    }

    @Override
    public List<Long> putItems(List<Store> stores) {
        return null;
    }

    @Override
    public Maybe<List<Long>> putItemsRx(List<Store> stores) {
        return null;
    }

    @Override
    public Store getItem(long id) {
        return null;
    }

    @Override
    public Maybe<Store> getItemRx(long id) {
        return null;
    }

    @Override
    public List<Store> getItems() {
        return null;
    }

    @Override
    public Maybe<List<Store>> getItemsRx() {
        return null;
    }

    @Override
    public List<Store> getItems(int limit) {
        return null;
    }

    @Override
    public Maybe<List<Store>> getItemsRx(int limit) {
        return null;
    }



}

File: frame/src/main/kotlin/com/dreampany/frame/data/source/repository/StoreRepository.kt
package com.dreampany.frame.data.source.repository

import com.dreampany.frame.data.enums.State
import com.dreampany.frame.data.enums.Subtype
import com.dreampany.frame.data.enums.Type
import com.dreampany.frame.data.model.Store
import com.dreampany.frame.data.source.api.StoreDataSource
import com.dreampany.frame.misc.ResponseMapper
import com.dreampany.frame.misc.Room
import com.dreampany.frame.misc.RxMapper
import io.reactivex.Maybe
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by roman on 2019-07-25
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
@Singleton
class StoreRepository
@Inject constructor(
    rx: RxMapper,
    rm: ResponseMapper,
    @Room private val room: StoreDataSource
) : Repository<String, Store>(rx, rm), StoreDataSource {

    override fun getItem(type: Type, subtype: Subtype, state: State): Store? {
        return room.getItem(type, subtype, state)
    }

    override fun getItems(type: Type, subtype: Subtype, state: State): List<Store>? {
        return room.getItems(type, subtype, state)
    }

    override fun getItemsRx(type: Type, subtype: Subtype, state: State): Maybe<List<Store>> {
        return room.getItemsRx(type, subtype, state)
    }

    override fun getCount(id: String, type: Type, subtype: Subtype, state: State): Int {
        return room.getCount(id, type, subtype, state)
    }

    override fun isExists(id: String, type: Type, subtype: Subtype, state: State): Boolean {
        return room.isExists(id, type, subtype, state)
    }

    override fun isExistsRx(
        id: String,
        type: Type,
        subtype: Subtype,
        state: State
    ): Maybe<Boolean> {
        return room.isExistsRx(id, type, subtype, state)
    }

    override fun getCount(id: String, type: Type, subtype: Subtype): Int {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getCount(): Int {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getCountRx(id: String, type: Type, subtype: Subtype): Maybe<Int> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getCountRx(): Maybe<Int> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getCountByType(type: Type, subtype: Subtype, state: State): Int {
        return room.getCountByType(type, subtype, state)
    }

    override fun getCountByTypeRx(type: Type, subtype: Subtype, state: State): Maybe<Int> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItem(id: String, type: Type, subtype: Subtype, state: State): Store? {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun putItem(t: Store): Long {
        return room.putItem(t)
    }

    override fun putItems(ts: List<Store>): List<Long>? {
        return room.putItems(ts)
    }

    override fun delete(t: Store): Int {
        return room.delete(t)
    }

    override fun deleteRx(t: Store): Maybe<Int> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItem(id: String): Store? {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItemRx(id: String, type: Type, subtype: Subtype, state: State): Maybe<Store> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun isEmpty(): Boolean {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun isEmptyRx(): Maybe<Boolean> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun isExists(t: Store): Boolean {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun isExistsRx(t: Store): Maybe<Boolean> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun putItemRx(t: Store): Maybe<Long> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun putItemsRx(ts: List<Store>): Maybe<List<Long>> {
        return room.putItemsRx(ts)
    }

    override fun delete(ts: List<Store>): List<Long>? {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun deleteRx(ts: List<Store>): Maybe<List<Long>> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItemRx(id: String): Maybe<Store> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItems(): List<Store>? {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItemsRx(): Maybe<List<Store>> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItems(limit: Int): List<Store>? {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItemsRx(limit: Int): Maybe<List<Store>> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }


}